### ISSUE TITLE
ImportError: cannot import name 'if_delegate_has_method' from 'sklearn.utils.metaestimators' with scikit-learn 1.3.0

### ISSUE THREAD

**Issue by RNarayan73** [manuel-calzolari/sklearn-genetic#42]
Hi All,

Looks like SKL 1.3.0 may have broken things!

**System information**
OS Platform and Distribution: Windows 11 Pro
Sklearn-genetic version: 0.5.1
deap version: 1.4.1
Scikit-learn version: 1.3.0
Python version: 3.10.12

**Describe the bug**
Merely importing GeneticSelectionCV results in an error as below:
> ImportError                               Traceback (most recent call last)
> Cell In[102], line 1
> ----> 1 from genetic_selection import GeneticSelectionCV
> 
> File ~\Miniconda3\envs\skl310\lib\site-packages\genetic_selection\__init__.py:16
>       1 # sklearn-genetic - Genetic feature selection module for scikit-learn
>       2 # Copyright (C) 2016-2022  Manuel Calzolari
>       3 #
>    (...)
>      13 # You should have received a copy of the GNU Lesser General Public License
>      14 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
> ---> 16 from .gscv import GeneticSelectionCV
>      18 __version__ = '0.5.1'
>      20 __all__ = ['GeneticSelectionCV']
> 
> File ~\Miniconda3\envs\skl310\lib\site-packages\genetic_selection\gscv.py:22
>      20 import numpy as np
>      21 from sklearn.utils import check_X_y
> ---> 22 from sklearn.utils.metaestimators import if_delegate_has_method
>      23 from sklearn.base import BaseEstimator
>      24 from sklearn.base import MetaEstimatorMixin
> 
> ImportError: cannot import name 'if_delegate_has_method' from 'sklearn.utils.metaestimators' (C:\Users\naray\Miniconda3\envs\skl310\lib\site-packages\sklearn\utils\metaestimators.py)
> 

**To Reproduce**
`from genetic_selection import GeneticSelectionCV`

**Expected behaviour**
The import shouldn't cause such an error.

**Additional context**

**Comment by manuel-calzolari** [comment]
Hi @RNarayan73, I fixed this issue with commit 12ee9b2e591ec379793bcff1966095b43dac10c6. I will release a new version of the package in the next few days after incorporating some other changes.

Duplicate of #37.

**Comment by RNarayan73** [comment]
@manuel-calzolari, still keenly awaiting release of this fix, which will hopefully address the related issue #16 too!

**Comment by manuel-calzolari** [comment]
@RNarayan73 
Sorry for the wait on the new release, things have been super busy on my end. I'm aiming to get it done by late December or early January.

**Comment by RNarayan73** [comment]
@manuel-calzolari 
Thanks for your effort. I really look forward to it.

**Comment by manuel-calzolari** [comment]
Fixed in version 0.6.0
